src/entry-server.js

```diff
@@ -23,8 +23,11 @@ export default function initializeServer(createApp, serverOpts) {
                 const components = router.getMatchedComponents();
 
                 if (!components.length) {
-                    opts.logger.warn(`No matched components for route: ${context.request.url}`);
-                    return reject({ code: 404, message: 'Not Found' });
+                    return reject({
+                        code: 404,
+                        message: 'Not Found',
+                        data: `No matched components for route: ${context.request.url}`,
+                    });
```

Comment:
Reduce the additional message and instead add the metadata to the rejected error